iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/ThreeWindingsTransformerAdapter.java

```diff
@@ -6,73 +6,99 @@
  */
 package com.powsybl.iidm.mergingview;
 
-import com.powsybl.iidm.network.ConnectableType;
-import com.powsybl.iidm.network.Substation;
-import com.powsybl.iidm.network.Terminal;
-import com.powsybl.iidm.network.ThreeWindingsTransformer;
+import com.powsybl.iidm.network.*;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
 /**
+ * This adaptation hide true implementation of {@link ThreeWindingsTransformer}.
+ *
  * @author Thomas Adam <tadam at silicom.fr>
  */
 public class ThreeWindingsTransformerAdapter extends AbstractIdentifiableAdapter<ThreeWindingsTransformer> implements ThreeWindingsTransformer {
 
+    private LegAdapter leg1;
+
+    private LegAdapter leg2;
+
+    private LegAdapter leg3;
+
     ThreeWindingsTransformerAdapter(final ThreeWindingsTransformer delegate, final MergingViewIndex index) {
         super(delegate, index);
     }
 
-    // -------------------------------
-    // Not implemented methods -------
-    // -------------------------------
     @Override
-    public ConnectableType getType() {
-        throw MergingView.NOT_IMPLEMENTED_EXCEPTION;
+    public ThreeWindingsTransformer.Leg getLeg1() {
+        // no need to store Leg1Adapter in MergingViewIndex
+        if (leg1 == null) {
+            leg1 = new LegAdapter(getDelegate().getLeg1(), getIndex());
+        }
+        return leg1;
```

Comment:
Could leg1 really be null? Maybe the first time you access to it? Same question for the other legs.

Reply:
Leg 1, 2 or 3 cannot be null
I will move Leg instantiations into the constructor